pkg/util/iptables/testing/fake.go

```diff
@@ -271,12 +271,11 @@ func (f *FakeIPTables) restoreTable(newDump *IPTablesDump, newTable *Table, flus
 					// was deleted by newDump.
 					oldTable.Chains = backupChains
 					return fmt.Errorf("deleted chain %q is referenced by existing rules", newChain.Name)
-				} else {
-					// rule is a new rule that jumped to a chain that was
-					// neither created nor pre-existing
-					oldTable.Chains = backupChains
-					return fmt.Errorf("rule %q jumps to a non-existent chain", rule.Raw)
 				}
+				// rule is a new rule that jumped to a chain that was
+				// neither created nor pre-existing
+				oldTable.Chains = backupChains
+				return fmt.Errorf("rule %q jumps to a non-existent chain", rule.Raw)
```

Comment:
I disagree with this change (and the similar changes to `pkg/util/iptables/testing/parse.go`; this isn't "if special case, do X, else do the normal thing". It's "There are exactly two possibilities; if it's the first possibility, do the first thing, else it's the second possibility, so do the second thing." Using an `else` represents the logical structure better in this case IMO.

Reply:
is a "go thing" as you can read it here https://go.dev/doc/effective_go#if

> In the Go libraries, you'll find that when an if statement doesn't flow into the next statement—that is, the body ends in break, continue, goto, or return—the unnecessary else is omitted.

Reply:
> I disagree with this change (and the similar changes to `pkg/util/iptables/testing/parse.go`; this isn't "if special case, do X, else do the normal thing". It's "There are exactly two possibilities; if it's the first possibility, do the first thing, else it's the second possibility, so do the second thing." Using an `else` represents the logical structure better in this case IMO.

I have canceled such changes

Reply:
> is a "go thing" as you can read it here https://go.dev/doc/effective_go#if

They don't distinguish "if special case, do X, else do the normal thing" from "if case 1 do thing 1 else do thing 2" in the recommendation, but the code examples they give are all the former, not the latter.

Reply:
what you mean is to keep the previous changes?